packages/toolpad-core/src/AppProvider/AppProvider.tsx

```diff
@@ -2,6 +2,8 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
 import { createTheme as createMuiTheme, Theme } from '@mui/material/styles';
+import { LinkProps as ReactRouterLinkProps } from 'react-router';
+import { LinkProps as NextLinkProps } from 'next/link';
```

Comment:
This seems incorrect because:
* the package should never import from third party runtimes
* I don't think the implementations are 100% compatible

We need to define an abstract LinkProps type ourselves, and then make sure in the routeradapter to map the concrete implementation to the abstract properties.